### PR TITLE
Replace CheckBoxList

### PR DESCRIPTION
**Pull request by YenguiSeddik**

### CODE REVIEW

File: src/components/graph/menus/network-modification-node-editor.jsx [hunk 1]
```diff
@@ -936,6 +958,21 @@ const NetworkModificationNodeEditor = () => {
...
+            {
+                ...modif,
+                ...computeLabel(modif),
+            }
+        );
+    };
+
     const renderNetworkModificationsList = () => {
         return (
             <DragDropContext
```
Comment (sBouzols) [inline src/components/graph/menus/network-modification-node-editor.jsx:978]: `DragDropContext`, `Droppable` should be moved in the CheckboxList component

File: src/components/dialogs/restore-node-dialog.jsx [hunk 2]
```diff
@@ -114,40 +89,16 @@ const RestoreNodesDialog = ({ open, onClose, anchorNodeId, studyUuid }) => {
...
-                    </FormControl>
+                    <CheckboxList
+                        items={nodes}
+                        addSelectAllCheckbox
+                        selectAllCheckBoxLabel={'SelectAll'}
+                        selectedItems={selectedNodes}
+                        onSelectionChange={setSelectedNodes}
+                        getItemId={(v) => v.first.id}
+                        getItemLabel={(v) => v.first.name + (v.second !== 0 ? ' ( + ' + v.second + ' )' : '')}
+                        divider
```
Comment (klesaulnier) [inline src/components/dialogs/restore-node-dialog.jsx:100]: I think there isn't any divider in main branch

File: src/components/graph/menus/dynamic-simulation/event-modification-scenario-editor.tsx [hunk 3]
```diff
@@ -198,23 +196,50 @@ const EventModificationScenarioEditor = () => {
...
+        }
+
+        const computedValues = {
+            computedLabel: (
+                <>
+                    <strong>{item.equipmentId}</strong>
+                    <i>{` - ${getStartTime(item)} ${getStartTimeUnit(item)}`}</i>
+                </>
+            ),
+        } as {};
```
Comment (klesaulnier) [inline src/components/graph/menus/dynamic-simulation/event-modification-scenario-editor.tsx:211]:
```suggestion
        };
```

**Review by klesaulnier — COMMENTED** [review]
left padding in checkbox list have increased, it is mostly visible in network modification node editor
the whole row is not clickable to select/unselect an element, instead of the checkbox only

some typescript errors, will end review when fixed

**Pull request merged**